modules/client-configuration/pages/configuration-management.adoc

```diff
@@ -1,88 +1,90 @@
 
-[[ref.webui.config.preparing]]
-== Prepare Systems for Configuration Management [Management]
+When you have created your configuration channels, you can also perform comparisons between various configuration files, and between revisions of the same configuration file.
 
-To manage traditional client configuration with {productname}, the client must have the appropriate tools and the [path]``config-enable`` file installed.
-These tools will be available if you installed the system with the configuration management functionality using {ay} or Kickstart.
-If not, they can be found in the Tools child channel for your distribution.
-Download and install the latest [path]``mgr-cfg*`` packages:
+Configuration files can be centrally managed, or locally managed.
+Centrally managed configuration files are provided by global configuration channels.
+Locally managed configuration files are created or uploaded to {productname} directly.
 
-* [path]``mgr-cfg`` -- the base libraries and functions needed by all [path]``mgr-cfg-*`` packages,
-* [path]``mgr-cfg-actions`` -- the RPM package required to run configuration actions scheduled via {productname},
-* [path]``mgr-cfg-client`` -- the RPM package with a command line interface to the client features of the Configuration Management system,
-* [path]``mgr-cfg-management`` -- the RPM package with a command line interface used to manage {productname} configuration.
 
+// Edited these for style, not tested. --LKB 2020-07-31
+== Prepare Traditional Clients for Configuration Management
 
-Installation of these packages can also be accomplished during bootstrapping if you enable [guimenu]``Configuration File Deployment`` on the [guimenu]``Details`` page of the activation key after creating that activation key.
+Traditional clients require some extra preparation to use configuration management.
+If you installed your traditional client with {ay} or {kickstart} you probably already have the appropriate packages.
+For other traditional clients, ensure you have installed the relevant tools child channel for your client operating system.
+For more information about software channels, see xref:client-configuration:channels.adoc[].
+
+The packages you require are:
+
+* [path]``mgr-cfg``: base libraries and functions needed by all [path]``mgr-cfg-*`` packages
+* [path]``mgr-cfg-actions``: required to run configuration actions scheduled via {productname}.
+* [path]``mgr-cfg-client``: provides a command line interface to the client features of the configuration management system.
+* [path]``mgr-cfg-management``: provides a command line interface to manage {productname} configuration.
+
+You can install these packages during the bootstrap process by navigating to menu:Systems[Activation Keys], clicking the activation key you intend to use during bootstrap, and checking the [guimenu]``Configuration File Deployment`` option.
```

Comment:
@0rnela @ktsamis could you confirm that in traditional clients, with that checkbox enabled, it will be installed during bootstrap?
There is a Tip in the checkbox, which says `Tip: If the system is registered via Salt, the highstate will be executed on registration if this checkbox is selected.` 
And I'm not sure if we might need to install it manually. What I remember is that for traditional clients, I use to manually install the `mgr-cfg-actions` when I run the remote commands

Reply:
Pretty sure at least `mgr-cfg-actions` is not installed but needs to be later. But the doc doesn't say they are automatically installed. It just says you need to have them installed.

Reply:
```You can install these packages during the bootstrap process by....```

Reply:
That should work, but you are specifying here extra packages, not the minimal set that would be done to be able to bootstrap.

Reply:
@Loquacity my understanding of that sentence `You can install these packages during the bootstrap process by navigating to menu:Systems[Activation Keys], clicking the activation key you intend to use during bootstrap, and checking the [guimenu]``Configuration File Deployment`` option.` is that it will all done automatically during the bootstrap process thanks to that checkbox option in the Activation key.  But here we have some doubts about some extra manual step to install extra packages. It would be good to test and validate this description. I'm gonna try it during 4.1.1 MU validation and bring back the outcome.

Reply:
OK, thanks Oscar!

Reply:
Confirmed. The bootstrap script for a sles15sp1 traditional client, having enabled the checkbox of `Configuration File Deployment` will not install any of the packages mentioned above: `mgr-cfg*`.

After bootstrap complete those are the commands starting by mgr available:
```
d314:~ # mgr
mgr-actions-control  mgr-update-status    mgr_check            mgrcfg-client   
```

Here the full bootstrap log:
``` 
d314:~ # bash bootstrap.sh 
SUSE Manager Server Client bootstrap script v4.1


MINOR MANUAL EDITING OF THIS FILE MAY BE REQUIRED!

If this bootstrap script was created during the initial installation
of a SUSE Manager Server, the ACTIVATION_KEYS, and ORG_GPG_KEY values will
probably *not* be set (see below). If this is the case, please do the
following:
  - copy this file to a name specific to its use.
    (e.g., to bootstrap-SOME_NAME.sh - like bootstrap-web-servers.sh.)
  - on the website create an activation key or keys for the system(s) to
    be registered.
  - edit the values of the VARIABLES below (in this script) as
    appropriate:
    - ACTIVATION_KEYS needs to reflect the activation key(s) value(s)
      from the website. XKEY or XKEY,YKEY
      Please note that if you are using this script to boostrap minions,
      only the FIRST activation key will be used. Multiple activation keys
      are not supported with salt
    - ORG_GPG_KEY needs to be set to the name(s) of the corporate public
      GPG key filename(s) (residing in /srv/www/htdocs/pub) if appropriate. XKEY or XKEY,YKEY

Verify that the script variable settings are correct:
    - CLIENT_OVERRIDES should be only set differently if a customized
      client-config-overrides-VER.txt file was created with a different
      name.
    - ensure the value of HOSTNAME is correct.
    - ensure the value of ORG_CA_CERT is correct.

Enable this script: comment (with #'s) this block (or, at least just
the exit below, if present)


UPDATING RHN_REGISTER/UP2DATE CONFIGURATION FILES
-------------------------------------------------
* downloading necessary files
  client_config_update.py...
  client-config-overrides.txt...

PREPARE GPG KEYS AND CORPORATE PUBLIC CA CERT
-------------------------------------------------
* no organizational GPG keys to import

* attempting to install corporate public CA cert
Preparing...                          ################################# [100%]
Updating / installing...
   1:rhn-org-trusted-ssl-cert-1.0-1   ################################# [100%]

CLEANING UP OLD SUSE MANAGER REPOSITORIES
-------------------------------------------------

CHECKING THE REGISTRATION STACK
-------------------------------------------------
* check for necessary packages being installed...
* client codebase is sle-15-sp1
package spacewalk-check is not installed
package spacewalk-client-setup is not installed
package spacewalk-client-tools is not installed
package zypp-plugin-spacewalk is not installed
* going to install missing packages...
 adding client software repository at https://mgr-srv-41.qam.suse.de/pub/repositories/sle/15/1/bootstrap
Retrieving repository 'susemanager:bootstrap' metadata .................................................................................................................................[done]
Building repository 'susemanager:bootstrap' cache ......................................................................................................................................[done]
Specified repositories have been refreshed.
Loading repository data...
Reading installed packages...
Resolving package dependencies...

The following 38 NEW packages are going to be installed:
  hwdata libgudev-1_0-0 libnewt0_52 libslang2 mgr-daemon newt python-dmidecode python3-appdirs python3-asn1crypto python3-cffi python3-cryptography python3-dmidecode python3-hwdata
  python3-idna python3-libxml2-python python3-netifaces python3-newt python3-packaging python3-pyOpenSSL python3-pyasn1 python3-pycparser python3-pyparsing python3-pyudev python3-rhnlib
  python3-rpm python3-setuptools python3-six python3-spacewalk-check python3-spacewalk-client-setup python3-spacewalk-client-tools python3-suseRegisterInfo python3-uyuni-common-libs
  python3-zypp-plugin-spacewalk spacewalk-check spacewalk-client-setup spacewalk-client-tools suseRegisterInfo zypp-plugin-spacewalk

The following 38 packages have no support information from their vendor:
  hwdata libgudev-1_0-0 libnewt0_52 libslang2 mgr-daemon newt python-dmidecode python3-appdirs python3-asn1crypto python3-cffi python3-cryptography python3-dmidecode python3-hwdata
  python3-idna python3-libxml2-python python3-netifaces python3-newt python3-packaging python3-pyOpenSSL python3-pyasn1 python3-pycparser python3-pyparsing python3-pyudev python3-rhnlib
  python3-rpm python3-setuptools python3-six python3-spacewalk-check python3-spacewalk-client-setup python3-spacewalk-client-tools python3-suseRegisterInfo python3-uyuni-common-libs
  python3-zypp-plugin-spacewalk spacewalk-check spacewalk-client-setup spacewalk-client-tools suseRegisterInfo zypp-plugin-spacewalk

38 new packages to install.
Overall download size: 5.2 MiB. Already cached: 0 B. After the operation, additional 27.1 MiB will be used.
Continue? [y/n/v/...? shows all options] (y): y
Retrieving package hwdata-0.334-3.12.1.noarch                                                                                                           (1/38),   1.3 MiB (  7.5 MiB unpacked)
Retrieving: hwdata-0.334-3.12.1.noarch.rpm .............................................................................................................................................[done]
Retrieving package libgudev-1_0-0-232-1.33.x86_64                                                                                                       (2/38),  28.1 KiB ( 68.9 KiB unpacked)
Retrieving: libgudev-1_0-0-232-1.33.x86_64.rpm .........................................................................................................................................[done]
Retrieving package libslang2-2.3.1a-3.13.x86_64                                                                                                         (3/38), 305.4 KiB (  1.1 MiB unpacked)
Retrieving: libslang2-2.3.1a-3.13.x86_64.rpm ...........................................................................................................................................[done]
Retrieving package python3-appdirs-1.4.3-1.21.noarch                                                                                                    (4/38),  22.5 KiB ( 83.5 KiB unpacked)
Retrieving: python3-appdirs-1.4.3-1.21.noarch.rpm ......................................................................................................................................[done]
Retrieving package python3-asn1crypto-0.24.0-1.20.noarch                                                                                                (5/38), 176.2 KiB (  1.2 MiB unpacked)
Retrieving: python3-asn1crypto-0.24.0-1.20.noarch.rpm ..................................................................................................................................[done]
Retrieving package python3-idna-2.6-1.20.noarch                                                                                                         (6/38),  85.0 KiB (772.7 KiB unpacked)
Retrieving: python3-idna-2.6-1.20.noarch.rpm ...........................................................................................................................................[done]
Retrieving package python3-libxml2-python-2.9.7-3.22.1.x86_64                                                                                           (7/38), 221.0 KiB (  1.6 MiB unpacked)
Retrieving: python3-libxml2-python-2.9.7-3.22.1.x86_64.rpm .............................................................................................................................[done]
Retrieving package python3-netifaces-0.10.6-1.31.x86_64                                                                                                 (8/38),  22.4 KiB ( 37.8 KiB unpacked)
Retrieving: python3-netifaces-0.10.6-1.31.x86_64.rpm ...................................................................................................................................[done]
Retrieving package python3-pyasn1-0.4.2-1.20.noarch                                                                                                     (9/38), 154.7 KiB (823.2 KiB unpacked)
Retrieving: python3-pyasn1-0.4.2-1.20.noarch.rpm .......................................................................................................................................[done]
Retrieving package python3-pycparser-2.17-1.24.noarch                                                                                                  (10/38), 189.9 KiB (  1.1 MiB unpacked)
Retrieving: python3-pycparser-2.17-1.24.noarch.rpm .......................................................................................................................[done (144.0 KiB/s)]
Retrieving package python3-pyparsing-2.2.0-1.28.noarch                                                                                                 (11/38), 158.8 KiB (723.3 KiB unpacked)
Retrieving: python3-pyparsing-2.2.0-1.28.noarch.rpm ....................................................................................................................................[done]
Retrieving package python3-rpm-4.14.1-10.19.18.x86_64                                                                                                  (12/38),  56.7 KiB (191.5 KiB unpacked)
Retrieving: python3-rpm-4.14.1-10.19.18.x86_64.rpm .....................................................................................................................................[done]
Retrieving package python3-six-1.11.0-4.3.2.noarch                                                                                                     (13/38),  32.2 KiB ( 91.0 KiB unpacked)
Retrieving: python3-six-1.11.0-4.3.2.noarch.rpm ........................................................................................................................................[done]
Retrieving package python3-suseRegisterInfo-4.1.2-3.6.2.noarch                                                                                         (14/38),  13.7 KiB ( 14.9 KiB unpacked)
Retrieving: python3-suseRegisterInfo-4.1.2-3.6.2.noarch.rpm ............................................................................................................................[done]
Retrieving package python3-uyuni-common-libs-4.1.5-1.3.2.x86_64                                                                                        (15/38),  55.0 KiB (160.4 KiB unpacked)
Retrieving: python3-uyuni-common-libs-4.1.5-1.3.2.x86_64.rpm ...........................................................................................................................[done]
Retrieving package python3-hwdata-2.3.5-1.21.noarch                                                                                                    (16/38),  36.2 KiB (175.5 KiB unpacked)
Retrieving: python3-hwdata-2.3.5-1.21.noarch.rpm .......................................................................................................................................[done]
Retrieving package libnewt0_52-0.52.20-5.35.x86_64                                                                                                     (17/38),  47.3 KiB ( 87.8 KiB unpacked)
Retrieving: libnewt0_52-0.52.20-5.35.x86_64.rpm ........................................................................................................................................[done]
Retrieving package python3-dmidecode-3.12.2-1.24.x86_64                                                                                                (18/38),  66.9 KiB (194.5 KiB unpacked)
Retrieving: python3-dmidecode-3.12.2-1.24.x86_64.rpm ...................................................................................................................................[done]
Retrieving package python3-cffi-1.11.2-4.3.1.x86_64                                                                                                    (19/38), 233.7 KiB (982.8 KiB unpacked)
Retrieving: python3-cffi-1.11.2-4.3.1.x86_64.rpm .......................................................................................................................................[done]
Retrieving package python3-pyudev-0.21.0-3.22.noarch                                                                                                   (20/38),  88.8 KiB (481.4 KiB unpacked)
Retrieving: python3-pyudev-0.21.0-3.22.noarch.rpm ......................................................................................................................................[done]
Retrieving package python3-packaging-16.8-1.23.noarch                                                                                                  (21/38),  41.8 KiB (161.9 KiB unpacked)
Retrieving: python3-packaging-16.8-1.23.noarch.rpm .....................................................................................................................................[done]
Retrieving package suseRegisterInfo-4.1.2-3.6.2.noarch                                                                                                 (22/38),   9.9 KiB (  1.8 KiB unpacked)
Retrieving: suseRegisterInfo-4.1.2-3.6.2.noarch.rpm ....................................................................................................................................[done]
Retrieving package newt-0.52.20-5.35.x86_64                                                                                                            (23/38),  51.5 KiB ( 91.1 KiB unpacked)
Retrieving: newt-0.52.20-5.35.x86_64.rpm ...............................................................................................................................................[done]
Retrieving package python-dmidecode-3.12.2-1.24.x86_64                                                                                                 (24/38),  23.7 KiB ( 73.2 KiB unpacked)
Retrieving: python-dmidecode-3.12.2-1.24.x86_64.rpm ....................................................................................................................................[done]
Retrieving package python3-setuptools-40.5.0-4.21.noarch                                                                                               (25/38), 615.8 KiB (  3.3 MiB unpacked)
Retrieving: python3-setuptools-40.5.0-4.21.noarch.rpm ..................................................................................................................................[done]
Retrieving package python3-newt-0.52.20-5.35.x86_64                                                                                                    (26/38),  38.5 KiB (140.2 KiB unpacked)
Retrieving: python3-newt-0.52.20-5.35.x86_64.rpm .......................................................................................................................................[done]
Retrieving package python3-cryptography-2.1.4-4.6.1.x86_64                                                                                             (27/38), 403.7 KiB (  2.4 MiB unpacked)
Retrieving: python3-cryptography-2.1.4-4.6.1.x86_64.rpm ................................................................................................................................[done]
Retrieving package python3-pyOpenSSL-17.5.0-3.6.1.noarch                                                                                               (28/38), 103.5 KiB (553.6 KiB unpacked)
Retrieving: python3-pyOpenSSL-17.5.0-3.6.1.noarch.rpm ..................................................................................................................................[done]
Retrieving package python3-rhnlib-4.1.2-3.16.2.noarch                                                                                                  (29/38),  66.7 KiB (238.6 KiB unpacked)
Retrieving: python3-rhnlib-4.1.2-3.16.2.noarch.rpm .....................................................................................................................................[done]
Retrieving package python3-spacewalk-client-tools-4.1.5-3.23.2.noarch                                                                                  (30/38), 101.7 KiB (429.1 KiB unpacked)
Retrieving: python3-spacewalk-client-tools-4.1.5-3.23.2.noarch.rpm .....................................................................................................................[done]
Retrieving package spacewalk-client-tools-4.1.5-3.23.2.noarch                                                                                          (31/38), 317.0 KiB (  1.9 MiB unpacked)
Retrieving: spacewalk-client-tools-4.1.5-3.23.2.noarch.rpm .............................................................................................................................[done]
Retrieving package zypp-plugin-spacewalk-1.0.7-3.12.2.noarch                                                                                           (32/38),  14.6 KiB ( 11.1 KiB unpacked)
Retrieving: zypp-plugin-spacewalk-1.0.7-3.12.2.noarch.rpm ..............................................................................................................................[done]
Retrieving package python3-zypp-plugin-spacewalk-1.0.7-3.12.2.noarch                                                                                   (33/38),  24.2 KiB ( 47.0 KiB unpacked)
Retrieving: python3-zypp-plugin-spacewalk-1.0.7-3.12.2.noarch.rpm ......................................................................................................................[done]
Retrieving package spacewalk-check-4.1.5-3.23.2.noarch                                                                                                 (34/38),  18.0 KiB (  2.7 KiB unpacked)
Retrieving: spacewalk-check-4.1.5-3.23.2.noarch.rpm ....................................................................................................................................[done]
Retrieving package python3-spacewalk-check-4.1.5-3.23.2.noarch                                                                                         (35/38),  29.5 KiB ( 42.1 KiB unpacked)
Retrieving: python3-spacewalk-check-4.1.5-3.23.2.noarch.rpm ............................................................................................................................[done]
Retrieving package mgr-daemon-4.1.1-1.14.2.noarch                                                                                                      (36/38),  27.5 KiB ( 19.7 KiB unpacked)
Retrieving: mgr-daemon-4.1.1-1.14.2.noarch.rpm .........................................................................................................................................[done]
Retrieving package python3-spacewalk-client-setup-4.1.5-3.23.2.noarch                                                                                  (37/38),  57.5 KiB (239.7 KiB unpacked)
Retrieving: python3-spacewalk-client-setup-4.1.5-3.23.2.noarch.rpm .....................................................................................................................[done]
Retrieving package spacewalk-client-setup-4.1.5-3.23.2.noarch                                                                                          (38/38),  19.5 KiB (  3.3 KiB unpacked)
Retrieving: spacewalk-client-setup-4.1.5-3.23.2.noarch.rpm .............................................................................................................................[done]
Checking for file conflicts: ...........................................................................................................................................................[done]
( 1/38) Installing: hwdata-0.334-3.12.1.noarch .........................................................................................................................................[done]
( 2/38) Installing: libgudev-1_0-0-232-1.33.x86_64 .....................................................................................................................................[done]
( 3/38) Installing: libslang2-2.3.1a-3.13.x86_64 .......................................................................................................................................[done]
( 4/38) Installing: python3-appdirs-1.4.3-1.21.noarch ..................................................................................................................................[done]
( 5/38) Installing: python3-asn1crypto-0.24.0-1.20.noarch ..............................................................................................................................[done]
( 6/38) Installing: python3-idna-2.6-1.20.noarch .......................................................................................................................................[done]
( 7/38) Installing: python3-libxml2-python-2.9.7-3.22.1.x86_64 .........................................................................................................................[done]
( 8/38) Installing: python3-netifaces-0.10.6-1.31.x86_64 ...............................................................................................................................[done]
( 9/38) Installing: python3-pyasn1-0.4.2-1.20.noarch ...................................................................................................................................[done]
(10/38) Installing: python3-pycparser-2.17-1.24.noarch .................................................................................................................................[done]
(11/38) Installing: python3-pyparsing-2.2.0-1.28.noarch ................................................................................................................................[done]
(12/38) Installing: python3-rpm-4.14.1-10.19.18.x86_64 .................................................................................................................................[done]
(13/38) Installing: python3-six-1.11.0-4.3.2.noarch ....................................................................................................................................[done]
(14/38) Installing: python3-suseRegisterInfo-4.1.2-3.6.2.noarch ........................................................................................................................[done]
(15/38) Installing: python3-uyuni-common-libs-4.1.5-1.3.2.x86_64 .......................................................................................................................[done]
(16/38) Installing: python3-hwdata-2.3.5-1.21.noarch ...................................................................................................................................[done]
(17/38) Installing: libnewt0_52-0.52.20-5.35.x86_64 ....................................................................................................................................[done]
(18/38) Installing: python3-dmidecode-3.12.2-1.24.x86_64 ...............................................................................................................................[done]
(19/38) Installing: python3-cffi-1.11.2-4.3.1.x86_64 ...................................................................................................................................[done]
(20/38) Installing: python3-pyudev-0.21.0-3.22.noarch ..................................................................................................................................[done]
(21/38) Installing: python3-packaging-16.8-1.23.noarch .................................................................................................................................[done]
(22/38) Installing: suseRegisterInfo-4.1.2-3.6.2.noarch ................................................................................................................................[done]
(23/38) Installing: newt-0.52.20-5.35.x86_64 ...........................................................................................................................................[done]
(24/38) Installing: python-dmidecode-3.12.2-1.24.x86_64 ................................................................................................................................[done]
(25/38) Installing: python3-setuptools-40.5.0-4.21.noarch ..............................................................................................................................[done]
Additional rpm output:
update-alternatives: using /usr/bin/easy_install-3.6 to provide /usr/bin/easy_install (easy_install) in auto mode


(26/38) Installing: python3-newt-0.52.20-5.35.x86_64 ...................................................................................................................................[done]
(27/38) Installing: python3-cryptography-2.1.4-4.6.1.x86_64 ............................................................................................................................[done]
(28/38) Installing: python3-pyOpenSSL-17.5.0-3.6.1.noarch ..............................................................................................................................[done]
(29/38) Installing: python3-rhnlib-4.1.2-3.16.2.noarch .................................................................................................................................[done]
(30/38) Installing: python3-spacewalk-client-tools-4.1.5-3.23.2.noarch .................................................................................................................[done]
(31/38) Installing: spacewalk-client-tools-4.1.5-3.23.2.noarch .........................................................................................................................[done]
(32/38) Installing: zypp-plugin-spacewalk-1.0.7-3.12.2.noarch ..........................................................................................................................[done]
(33/38) Installing: python3-zypp-plugin-spacewalk-1.0.7-3.12.2.noarch ..................................................................................................................[done]
(34/38) Installing: spacewalk-check-4.1.5-3.23.2.noarch ................................................................................................................................[done]
(35/38) Installing: python3-spacewalk-check-4.1.5-3.23.2.noarch ........................................................................................................................[done]
(36/38) Installing: mgr-daemon-4.1.1-1.14.2.noarch .....................................................................................................................................[done]
Additional rpm output:
Created symlink /etc/systemd/system/timers.target.wants/rhnsd.timer -> /usr/lib/systemd/system/rhnsd.timer.
Created symlink /etc/systemd/system/multi-user.target.wants/spacewalk-update-status.service -> /usr/lib/systemd/system/spacewalk-update-status.service.


(37/38) Installing: python3-spacewalk-client-setup-4.1.5-3.23.2.noarch .................................................................................................................[done]
(38/38) Installing: spacewalk-client-setup-4.1.5-3.23.2.noarch .........................................................................................................................[done]
Executing %posttrans scripts ...........................................................................................................................................................[done]
spacewalk-check-4.1.5-3.23.2.noarch
spacewalk-client-setup-4.1.5-3.23.2.noarch
spacewalk-client-tools-4.1.5-3.23.2.noarch
zypp-plugin-spacewalk-1.0.7-3.12.2.noarch
Refreshing service 'spacewalk'.
Problem retrieving the repository index file for service 'spacewalk':
[spacewalk|file:/usr/lib/zypp/plugins/services/spacewalk] This system is not registered to any spacewalk server. If the system is not intended to be managed with spacewalk, please uninstall the zypp-plugin-spacewalk package.

Warning: Skipping service 'spacewalk' because of the above error.
Loading repository data...
Reading installed packages...
No update candidate for 'spacewalk-client-tools-4.1.5-3.23.2.noarch'. The highest available version is already installed.
No update candidate for 'python3-rhnlib-4.1.2-3.16.2.noarch'. The highest available version is already installed.
No update candidate for 'zypp-plugin-spacewalk-1.0.7-3.12.2.noarch'. The highest available version is already installed.
No update candidate for 'spacewalk-check-4.1.5-3.23.2.noarch'. The highest available version is already installed.
No update candidate for 'spacewalk-client-setup-4.1.5-3.23.2.noarch'. The highest available version is already installed.
'zypper' is already installed.
Package 'zypper' is not available in your repositories. Cannot reinstall, upgrade, or downgrade.
'openssl' is already installed.
Package 'openssl' is not available in your repositories. Cannot reinstall, upgrade, or downgrade.
Resolving package dependencies...

Nothing to do.
* running the update scripts
  . up2date config file
* /etc/sysconfig/rhn/up2date written

REGISTRATION
------------
* registering

*** this system should now be registered, please verify ***


* setting permissions to allow configuration management
  NOTE: use an activation key to subscribe to the tools
        channel and zypper install/update rhncfg-actions
        or zypper install/update mgr-cfg-actions starting with 4.0

* setting permissions to allow remote commands
  NOTE: use an activation key to subscribe to the tools
        channel and zypper update rhncfg-actions
        or zypper update mgr-cfg-actions starting with 4.0

OTHER ACTIONS
------------------------------------------------------
* Disable all repos not provided by SUSE Manager Server.
Error while modifying the service:
[spacewalk|file:/usr/lib/zypp/plugins/services/spacewalk] Service plugin does not support changing an attribute.
Leaving service spacewalk unchanged.
Nothing to change for repository 'SLES15-SP1-15.1-0'.
Nothing to change for repository 'SLES15-SP1-15.1-0'.
Repository 'spacewalk:sle-manager-tools15-pool-x86_64-sp1' has been successfully disabled. [volatile]
Warning: Repo 'spacewalk:sle-manager-tools15-pool-x86_64-sp1' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-manager-tools15-updates-x86_64-sp1' has been successfully disabled. [volatile]
Warning: Repo 'spacewalk:sle-manager-tools15-updates-x86_64-sp1' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-module-basesystem15-sp1-pool-x86_64' has been successfully disabled. [volatile]
Warning: Repo 'spacewalk:sle-module-basesystem15-sp1-pool-x86_64' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-module-basesystem15-sp1-updates-x86_64' has been successfully disabled. [volatile]
Warning: Repo 'spacewalk:sle-module-basesystem15-sp1-updates-x86_64' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-module-server-applications15-sp1-pool-x86_64' has been successfully disabled. [volatile]
Warning: Repo 'spacewalk:sle-module-server-applications15-sp1-pool-x86_64' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-module-server-applications15-sp1-updates-x86_64' has been successfully disabled. [volatile]
Warning: Repo 'spacewalk:sle-module-server-applications15-sp1-updates-x86_64' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-product-sles15-sp1-pool-x86_64' has been successfully disabled. [volatile]
Warning: Repo 'spacewalk:sle-product-sles15-sp1-pool-x86_64' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-product-sles15-sp1-updates-x86_64' has been successfully disabled. [volatile]
Warning: Repo 'spacewalk:sle-product-sles15-sp1-updates-x86_64' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle15sp1_customs' has been successfully disabled. [volatile]
Warning: Repo 'spacewalk:sle15sp1_customs' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-manager-tools15-pool-x86_64-sp1' has been successfully enabled. [volatile]
Warning: Repo 'spacewalk:sle-manager-tools15-pool-x86_64-sp1' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-manager-tools15-updates-x86_64-sp1' has been successfully enabled. [volatile]
Warning: Repo 'spacewalk:sle-manager-tools15-updates-x86_64-sp1' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-module-basesystem15-sp1-pool-x86_64' has been successfully enabled. [volatile]
Warning: Repo 'spacewalk:sle-module-basesystem15-sp1-pool-x86_64' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-module-basesystem15-sp1-updates-x86_64' has been successfully enabled. [volatile]
Warning: Repo 'spacewalk:sle-module-basesystem15-sp1-updates-x86_64' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-module-server-applications15-sp1-pool-x86_64' has been successfully enabled. [volatile]
Warning: Repo 'spacewalk:sle-module-server-applications15-sp1-pool-x86_64' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-module-server-applications15-sp1-updates-x86_64' has been successfully enabled. [volatile]
Warning: Repo 'spacewalk:sle-module-server-applications15-sp1-updates-x86_64' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-product-sles15-sp1-pool-x86_64' has been successfully enabled. [volatile]
Warning: Repo 'spacewalk:sle-product-sles15-sp1-pool-x86_64' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle-product-sles15-sp1-updates-x86_64' has been successfully enabled. [volatile]
Warning: Repo 'spacewalk:sle-product-sles15-sp1-updates-x86_64' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository 'spacewalk:sle15sp1_customs' has been successfully enabled. [volatile]
Warning: Repo 'spacewalk:sle15sp1_customs' is managed by service 'spacewalk'. Volatile changes are reset by the next service refresh!
Repository susemanager:bootstrap not found.
zypper --non-interactive up zypper zypp-plugin-spacewalk; rhn-profile-sync
but any post configuration action can be added here.  
------------------------------------------------------
* ensuring zypper itself is updated
Refreshing service 'spacewalk'.
Repository priorities are without effect. All enabled repositories share the same priority.

#  | Alias                                                         | Name                                                    | Enabled | GPG Check | Refresh | URI                                                                                  
---+---------------------------------------------------------------+---------------------------------------------------------+---------+-----------+---------+--------------------------------------------------------------------------------------
 1 | SLES15-SP1-15.1-0                                             | SLES15-SP1-15.1-0                                       | No      | ----      | ----    | cd:/?devices=/dev/disk/by-id/ata-QEMU_DVD-ROM_QM00002                                
 2 | spacewalk:sle-manager-tools15-pool-x86_64-sp1                 | SLE-Manager-Tools15-Pool for x86_64 SP1                 | Yes     | ( p) Yes  | Yes     | plugin:spacewalk?channel=sle-manager-tools15-pool-x86_64-sp1&server=0                
 3 | spacewalk:sle-manager-tools15-updates-x86_64-sp1              | SLE-Manager-Tools15-Updates for x86_64 SP1              | Yes     | ( p) Yes  | Yes     | plugin:spacewalk?channel=sle-manager-tools15-updates-x86_64-sp1&server=0             
 4 | spacewalk:sle-module-basesystem15-sp1-pool-x86_64             | SLE-Module-Basesystem15-SP1-Pool for x86_64             | Yes     | ( p) Yes  | Yes     | plugin:spacewalk?channel=sle-module-basesystem15-sp1-pool-x86_64&server=0            
 5 | spacewalk:sle-module-basesystem15-sp1-updates-x86_64          | SLE-Module-Basesystem15-SP1-Updates for x86_64          | Yes     | ( p) Yes  | Yes     | plugin:spacewalk?channel=sle-module-basesystem15-sp1-updates-x86_64&server=0         
 6 | spacewalk:sle-module-server-applications15-sp1-pool-x86_64    | SLE-Module-Server-Applications15-SP1-Pool for x86_64    | Yes     | ( p) Yes  | Yes     | plugin:spacewalk?channel=sle-module-server-applications15-sp1-pool-x86_64&server=0   
 7 | spacewalk:sle-module-server-applications15-sp1-updates-x86_64 | SLE-Module-Server-Applications15-SP1-Updates for x86_64 | Yes     | ( p) Yes  | Yes     | plugin:spacewalk?channel=sle-module-server-applications15-sp1-updates-x86_64&server=0
 8 | spacewalk:sle-product-sles15-sp1-pool-x86_64                  | SLE-Product-SLES15-SP1-Pool for x86_64                  | Yes     | ( p) Yes  | Yes     | plugin:spacewalk?channel=sle-product-sles15-sp1-pool-x86_64&server=0                 
 9 | spacewalk:sle-product-sles15-sp1-updates-x86_64               | SLE-Product-SLES15-SP1-Updates for x86_64               | Yes     | ( p) Yes  | Yes     | plugin:spacewalk?channel=sle-product-sles15-sp1-updates-x86_64&server=0              
10 | spacewalk:sle15sp1_customs                                    | sle15sp1_customs                                        | Yes     | ( p) Yes  | Yes     | plugin:spacewalk?channel=sle15sp1_customs&server=0                                   
Refreshing service 'spacewalk'.
All services have been refreshed.
Retrieving repository 'SLE-Manager-Tools15-Pool for x86_64 SP1' metadata ...............................................................................................................[done]
Retrieving repository 'SLE-Manager-Tools15-Updates for x86_64 SP1' metadata ............................................................................................................[done]
Retrieving repository 'SLE-Module-Basesystem15-SP1-Pool for x86_64' metadata ...........................................................................................................[done]
Retrieving repository 'SLE-Module-Basesystem15-SP1-Updates for x86_64' metadata ........................................................................................................[done]
Retrieving repository 'SLE-Module-Server-Applications15-SP1-Pool for x86_64' metadata ..................................................................................................[done]
Retrieving repository 'SLE-Module-Server-Applications15-SP1-Updates for x86_64' metadata ...............................................................................................[done]
Retrieving repository 'SLE-Product-SLES15-SP1-Pool for x86_64' metadata ................................................................................................................[done]
Retrieving repository 'SLE-Product-SLES15-SP1-Updates for x86_64' metadata .............................................................................................................[done]
Retrieving repository 'sle15sp1_customs' metadata ......................................................................................................................................[done]
All repositories have been refreshed.
Refreshing service 'spacewalk'.
Loading repository data...
Reading installed packages...
No update candidate for 'zypp-plugin-spacewalk-1.0.7-3.12.2.noarch'. The highest available version is already installed.
Resolving package dependencies...

The following 4 NEW packages are going to be installed:
  libsigc-2_0-0 libzstd1 lsof zypper-log

The following 3 packages are going to be upgraded:
  libsolv-tools libzypp zypper

3 packages to upgrade, 4 new.
Overall download size: 5.0 MiB. Already cached: 0 B. After the operation, additional 2.2 MiB will be used.
Continue? [y/n/v/...? shows all options] (y): y
Retrieving package lsof-4.91-1.11.x86_64                                                                                                                (1/7), 330.9 KiB (1007.1 KiB unpacked)
Retrieving: lsof-4.91-1.11.x86_64.rpm ..................................................................................................................................................[done]
Retrieving package libzstd1-1.4.4-1.3.1.x86_64                                                                                                           (2/7), 249.9 KiB (666.2 KiB unpacked)
Retrieving: libzstd1-1.4.4-1.3.1.x86_64.rpm ............................................................................................................................................[done]
Retrieving package libsigc-2_0-0-2.10.0-3.5.1.x86_64                                                                                                     (3/7),  74.7 KiB (269.7 KiB unpacked)
Retrieving: libsigc-2_0-0-2.10.0-3.5.1.x86_64.rpm ......................................................................................................................................[done]
Retrieving package zypper-log-1.14.37-3.19.1.noarch                                                                                                      (4/7),  46.2 KiB (  6.2 KiB unpacked)
Retrieving: zypper-log-1.14.37-3.19.1.noarch.rpm .......................................................................................................................................[done]
Retrieving package libsolv-tools-0.7.14-3.22.2.x86_64                                                                                                    (5/7), 630.5 KiB (  5.0 MiB unpacked)
Retrieving: libsolv-tools-0.7.14-3.22.2.x86_64.rpm .....................................................................................................................................[done]
Retrieving package libzypp-17.24.1-3.28.1.x86_64                                                                                                         (6/7),   2.2 MiB (  8.1 MiB unpacked)
Retrieving: libzypp-17.24.1-3.28.1.x86_64.rpm ..........................................................................................................................................[done]
Retrieving package zypper-1.14.37-3.19.1.x86_64                                                                                                          (7/7),   1.5 MiB (  7.5 MiB unpacked)
Retrieving: zypper-1.14.37-3.19.1.x86_64.rpm ...........................................................................................................................................[done]
Checking for file conflicts: ...........................................................................................................................................................[done]
(1/7) Installing: lsof-4.91-1.11.x86_64 ................................................................................................................................................[done]
(2/7) Installing: libzstd1-1.4.4-1.3.1.x86_64 ..........................................................................................................................................[done]
(3/7) Installing: libsigc-2_0-0-2.10.0-3.5.1.x86_64 ....................................................................................................................................[done]
(4/7) Installing: zypper-log-1.14.37-3.19.1.noarch .....................................................................................................................................[done]
(5/7) Installing: libsolv-tools-0.7.14-3.22.2.x86_64 ...................................................................................................................................[done]
(6/7) Installing: libzypp-17.24.1-3.28.1.x86_64 ........................................................................................................................................[done]
(7/7) Installing: zypper-1.14.37-3.19.1.x86_64 .........................................................................................................................................[done]
 
Updating package profile...
Updating hardware profile...
-bootstrap complete-
```

Reply:
OK, so the correct packages are listed, and they are installed by following the instructions, that's good! Do you think we need to list the commands it enables, or not?